### PR TITLE
More fully-functional `team list-memberships`

### PR DESCRIPTION
**Pull request by patrickxb**

### CODE REVIEW

File: protocol/avdl/keybase1/teams.avdl [hunk 1]
```diff
@@ -229,8 +229,30 @@ protocol teams {
     boolean staleOK;         // If a very stale cache hit is OK.
   }
 
+  record ImplicitRole {
+    TeamRole role;
+    TeamID ancestor;
+  }
+
+  record MemberInfo {
+    @jsonkey("team_id")
```
Comment (mmaxim) [inline protocol/avdl/keybase1/teams.avdl]: What does this do?

Comment (patrickxb) [inline protocol/avdl/keybase1/teams.avdl]: sets the json field in the go struct to use `team_id`.

**Comment by patrickxb** [comment]
I should mention that the `--all` flag isn't implemented.  A new server endpoint would make that a lot more efficient, but it can be done with existing endpoints if we need to.

File: protocol/avdl/keybase1/teams.avdl (hunk 1 above)
Comment (mmaxim) [inline protocol/avdl/keybase1/teams.avdl]: Ah gotcha, thanks!

**Pull request merged**